File: app/src/main/java/com/example/appswim/models/CreateSwimmerActivity.java
package com.example.appswim.models;

import android.arch.persistence.room.Room;
import android.database.sqlite.SQLiteConstraintException;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.example.appswim.R;
import com.example.appswim.Swimmer;
import com.example.appswim.database.SwimmerDao;
import com.example.appswim.database.SwimmerDatabase;


public class CreateSwimmerActivity extends AppCompatActivity {
    private EditText mFirstNameEditText;
    private EditText mLastNameEditText;
    private EditText mGenreEditText;
    private EditText mAgeEditText;
    private EditText mHeightEditText;
    private EditText mWeightEditText;
    private EditText mTeamEditText;

    private Button mSaveButton;
    private SwimmerDao mSwimmerDao;

    @Override

    protected  void onCreate(Bundle saveInstanceState) {
        super.onCreate(saveInstanceState);
        setContentView(R.layout.activity_create_swimmer);

        mSwimmerDao = Room.databaseBuilder(this, SwimmerDatabase.class, "db-swimmer")
                .allowMainThreadQueries()
                .build()
                .getswimmerDao();

        mFirstNameEditText = findViewById(R.id.firstNameEditText);
        mLastNameEditText = findViewById(R.id.lastNameEditText);
        mGenreEditText = findViewById(R.id.genreEditText);
        mAgeEditText = findViewById(R.id.ageEditText);
        mHeightEditText = findViewById(R.id.heightEditText);
        mWeightEditText = findViewById(R.id.weightEditText);
        mTeamEditText = findViewById(R.id.teamEditText);

        mSaveButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String m_first_name = mFirstNameEditText.getText().toString();
                String m_last_name = mLastNameEditText.getText().toString();
                String m_genre = mGenreEditText.getText().toString();
                String m_age = mAgeEditText.getText().toString();
                String m_height = mHeightEditText.getText().toString();
                String m_weight = mWeightEditText.getText().toString();
                String m_team = mTeamEditText.getText().toString();

                if (m_first_name.length() == 0 || m_last_name.length() == 0 || m_genre.length() == 0
                || m_age.length() == 0 || m_height.length() == 0 || m_weight.length() == 0 ||
                        m_team.length() == 0){
                    Toast.makeText(CreateSwimmerActivity.this, "Please make sure all details are correct", Toast.LENGTH_SHORT).show();
                    return;
                }

                Swimmer swimmer = new Swimmer();
                swimmer.setFirst_name(m_first_name);
                swimmer.setLast_name(m_last_name);
                swimmer.setGenre(m_genre);
                swimmer.setAge(m_age);
                swimmer.setHeight(m_height);
                swimmer.setWeight(m_weight);
                swimmer.setTeam(m_team);

                try{
                    mSwimmerDao.addSwimmer(swimmer);
                    setResult(RESULT_OK);
                    finish();
                }catch (SQLiteConstraintException e) {
                    Toast.makeText(CreateSwimmerActivity.this, "A swimmer already exists.", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: app/src/main/java/com/example/appswim/CreateSwimmerActivity.kt
package com.example.appswim


import android.app.Activity
import android.app.DatePickerDialog
import android.arch.persistence.room.Room
import android.database.sqlite.SQLiteConstraintException
import android.icu.util.Calendar
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

import com.example.appswim.database.SwimmerDao
import com.example.appswim.database.SwimmerDatabase
import com.example.appswim.models.Swimmer
import kotlinx.android.synthetic.main.activity_create_swimmer.*


class CreateSwimmerActivity : AppCompatActivity() {

    private var mFirstNameEditText: EditText? = null
    private var mLastNameEditText: EditText? = null
    private var mPhoneNumberEditText: EditText? = null
    private var mGenreEditText: EditText? = null
    private var mHeightEditText: EditText? = null
    private var mWeightEditText: EditText? = null
    private var mTeamEditText: EditText? = null
    private var mSaveButton: Button? = null
    private var mSwimmerDao: SwimmerDao? = null

    override fun onCreate(saveInstanceState: Bundle?) {
        super.onCreate(saveInstanceState)
        setContentView(R.layout.activity_create_swimmer)

        // Calendario

        val c = Calendar.getInstance()
        val day = c.get(Calendar.DAY_OF_MONTH)
        val month = c.get(Calendar.MONTH)
        val year = c.get(Calendar.YEAR)

        btnClickDate.setOnClickListener {
                val dpd = DatePickerDialog(this,android.R.style.Theme_Holo_Light_Dialog,
                    DatePickerDialog.OnDateSetListener{view, mYear,mMonth,
                                                       mDay->dateTv.text =
                            ("My birthday is "+mDay+"/"+mMonth + "/" + mYear)}, year,month,day)
            dpd.show()
        }

        mSwimmerDao = Room.databaseBuilder(this, SwimmerDatabase::class.java, "db-swimmer")
                .allowMainThreadQueries()
                .build()
                .swimmerDao

        mFirstNameEditText = findViewById(R.id.firstNameEditText)
        mLastNameEditText = findViewById(R.id.lastNameEditText)
        mGenreEditText = findViewById(R.id.genreEditText)
        mPhoneNumberEditText = findViewById(R.id.phoneNumberEditText)
        mHeightEditText = findViewById(R.id.heightEditText)
        mWeightEditText = findViewById(R.id.weightEditText)
        mTeamEditText = findViewById(R.id.teamEditText)
        mSaveButton = findViewById(R.id.saveButton)

        mSaveButton!!.setOnClickListener(View.OnClickListener {
            val FirstName = mFirstNameEditText!!.text.toString()
            val LastName = mLastNameEditText!!.text.toString()
            val Genre = mGenreEditText!!.text.toString()
            val PhoneNumber = mPhoneNumberEditText!!.text.toString()
            val Age = dateTv.text.toString()
            val Height = mHeightEditText!!.text.toString()
            val Weight = mWeightEditText!!.text.toString()
            val Team = mTeamEditText!!.text.toString()

            if(FirstName.length == 0){Toast.makeText(this@CreateSwimmerActivity,
                    "Please complete Firstname", Toast.LENGTH_SHORT).show()
                return@OnClickListener}
            else if(LastName.length == 0) {Toast.makeText(this@CreateSwimmerActivity,
                    "Please complete LastName", Toast.LENGTH_SHORT).show()
                return@OnClickListener}
            else if(PhoneNumber.length == 0){Toast.makeText(this@CreateSwimmerActivity,
                    "Please complete PhoneNumber", Toast.LENGTH_SHORT).show()
                return@OnClickListener}
            else if (Age.length == 0){Toast.makeText(this@CreateSwimmerActivity,
                    "Please select Age", Toast.LENGTH_SHORT).show()
                return@OnClickListener}
            else if(Height.length == 0){Toast.makeText(this@CreateSwimmerActivity,
                    "Please complete Height", Toast.LENGTH_SHORT).show()
                return@OnClickListener}
            else if(Weight.length == 0){Toast.makeText(this@CreateSwimmerActivity,
                    "Please complete Weight", Toast.LENGTH_SHORT).show()
                return@OnClickListener}
            else if(Team.length == 0){Toast.makeText(this@CreateSwimmerActivity,
                    "Please complete Team", Toast.LENGTH_SHORT).show()
                return@OnClickListener}

            val swimmer = Swimmer(
                    FirstName = FirstName,
                    LastName = LastName,
                    Genre = Genre,
                    PhoneNumber = PhoneNumber,
                    Age = Age,
                    Height = Height,
                    Weight = Weight,
                    Team = Team
            )

            try {
                mSwimmerDao!!.addSwimmer(swimmer)
                setResult(Activity.RESULT_OK)
                finish()
            } catch (e: SQLiteConstraintException) {
                Toast.makeText(this@CreateSwimmerActivity, "A swimmer already exists.", Toast.LENGTH_SHORT).show()
            }
        })
    }
}



